cli/cmd/validate.go

```diff
@@ -0,0 +1,135 @@
+	"github.com/spf13/cobra"
+)
+
+const (
+	validateShort   = "Validate spec(s) files"
+	validateExample = `# Validate spec(s) files
+cloudquery validate ./directory
+# Validate specs from directories and files
+cloudquery validate ./directory ./aws.yml ./pg.yml
+`
+)
+
+func newCmdValidate() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "validate [files or directories]",
+		Short:   validateShort,
+		Long:    validateShort,
+		Example: validateExample,
+		Args:    cobra.MinimumNArgs(1),
+		RunE:    validate,
+		Hidden:  true,
+	}
+	return cmd
+}
+
+func validate(cmd *cobra.Command, args []string) error {
+	cqDir, err := cmd.Flags().GetString("cq-dir")
+	if err != nil {
+		return err
+	}
+
+	ctx := cmd.Context()
+	log.Info().Strs("args", args).Msg("Loading spec(s)")
+	fmt.Printf("Loading spec(s) from %s\n", strings.Join(args, ", "))
+	specReader, err := specs.NewSpecReader(args)
+	if err != nil {
+		return fmt.Errorf("failed to load spec(s) from %s. Error: %w", strings.Join(args, ", "), err)
+	}
+	sources := specReader.Sources
+	destinations := specReader.Destinations
+
+	authToken, err := auth.GetAuthTokenIfNeeded(log.Logger, sources, destinations)
+	if err != nil {
+		return fmt.Errorf("failed to get auth token: %w", err)
+	}
+	teamName, err := auth.GetTeamForToken(authToken)
+	if err != nil {
+		return fmt.Errorf("failed to get team name: %w", err)
+	}
+	opts := []managedplugin.Option{
+		managedplugin.WithLogger(log.Logger),
+		managedplugin.WithAuthToken(authToken.Value),
+		managedplugin.WithTeamName(teamName),
+	}
+	if cqDir != "" {
+		opts = append(opts, managedplugin.WithDirectory(cqDir))
+	}
+	if disableSentry {
+		opts = append(opts, managedplugin.WithNoSentry())
+	}
+
+	sourcePluginConfigs := make([]managedplugin.Config, len(sources))
+	sourceRegInferred := make([]bool, len(sources))
+	for i, source := range sources {
+		sourcePluginConfigs[i] = managedplugin.Config{
+			Name:       source.Name,
+			Version:    source.Version,
+			Path:       source.Path,
+			Registry:   SpecRegistryToPlugin(source.Registry),
+			DockerAuth: source.DockerRegistryAuthToken,
+		}
+		sourceRegInferred[i] = source.RegistryInferred()
+	}
+	destinationPluginConfigs := make([]managedplugin.Config, len(destinations))
+	destinationRegInferred := make([]bool, len(destinations))
+	for i, destination := range destinations {
+		destinationPluginConfigs[i] = managedplugin.Config{
+			Name:       destination.Name,
+			Version:    destination.Version,
+			Path:       destination.Path,
+			Registry:   SpecRegistryToPlugin(destination.Registry),
+			DockerAuth: destination.DockerRegistryAuthToken,
+		}
+		destinationRegInferred[i] = destination.RegistryInferred()
+	}
+
+	sourceClients, err := managedplugin.NewClients(ctx, managedplugin.PluginSource, sourcePluginConfigs, opts...)
+	if err != nil {
+		return enrichClientError(sourceClients, sourceRegInferred, err)
+	}
+	defer func() {
+		if err := sourceClients.Terminate(); err != nil {
+			fmt.Println(err)
```

Comment:
Can we use the logger?

Reply:
This is similar to https://github.com/cloudquery/cloudquery/blob/c6e06aa2fb25010e0266538200dcb7212aa17248/cli/cmd/sync.go#L163, where we want to show the error in `stdout`